test/performance/io/format_fastq_benchmark.cpp

```diff
@@ -0,0 +1,256 @@
+{
+    using namespace seqan;
+
+    std::istringstream istream{};
+    auto file = get_file(n_entries_in_file);
+
+    auto restart_iterator = [&istream, &file]()    // c.f. format_fasta_benchmark
+    {
+        istream = std::istringstream{file};
+        // same constant as seqan3 benchmark
+
+        seqan::VirtualStream<char, Input> comp{};
+        open(comp, istream);
+        return directionIterator(comp, Input());
+    };
+
+    String<char> id{};
+    String<Dna5> seq{};
+    String<char> qual{};
+
+    for (auto _ : state)
+    {
+        auto it = restart_iterator();
+
+        for (size_t i = 0; i < n_entries_in_file; ++i)
+            readRecord(id, seq, qual, it, seqan::Fastq{});
+
+        clear(id);
+        clear(seq);
+        clear(qual);
+    }
+}
+
+#endif
+
+// ============================================================================
+// read dummy fastq file from temporary file on disk
+// ============================================================================
+template<size_t n_entries_in_file>
+void fastq_read_from_disk(benchmark::State & state)
+{
+    sequence_file_format_fastq format;
+
+    // create temporary file, automatically removed on destruction
+    test::tmp_filename file_name{"tmp.fastq"};
+    auto tmp_path = file_name.get_path();
+
+    std::ofstream ostream{tmp_path};
+    ostream << get_file(n_entries_in_file);
+    ostream.close();
+
+    // benchmark
+    std::string id{};
+    std::vector<dna5> seq{};
+    std::vector<phred42> qual{};
+
+    for (auto _ : state)
+    {
+        sequence_file_input fin{tmp_path};
+
+        // read all records and store in internal buffer
+        auto it = fin.begin();
+        while (it != fin.end())
+            ++it;
+    }
+}
+
+// ============================================================================
+// seqan2 comparison
+// ============================================================================
+#if SEQAN3_HAS_SEQAN2
+
+template<size_t n_entries_in_file>
+void fastq_read_from_disk_seqan2(benchmark::State & state)
+{
+    using namespace seqan;
+
+    // temporary file
+    test::tmp_filename file_name{"tmp.fastq"};
+    auto tmp_path = file_name.get_path();
+
+    std::ofstream ostream{tmp_path};
+    ostream << get_file(n_entries_in_file);
+    ostream.close();
+
+    // benchmark
+    StringSet<String<char>> ids{};
+    StringSet<String<Dna5>> seqs{};
+    StringSet<String<char>> quals{};
+
+    for (auto _ : state)
+    {
+        SeqFileIn seqFileIn(tmp_path.c_str());
+        readRecords(ids, seqs, quals, seqFileIn);
+
+        clear(ids);
+        clear(seqs);
+        clear(quals);
+    }
+}
+
+#endif
+
+// weirdly formatted code but makes console output easier to read
+BENCHMARK(fastq_write_to_stream);
+
+BENCHMARK_TEMPLATE(fastq_read_from_stream, 100);
+BENCHMARK_TEMPLATE(fastq_read_from_disk, 100);
+
+#if SEQAN3_HAS_SEQAN2
+    BENCHMARK_TEMPLATE(fastq_read_from_stream_seqan2, 100);
```

Comment:
We normally don't add a level of indentation within `#if ... #endif` guards.

```suggestion
BENCHMARK_TEMPLATE(fastq_read_from_stream_seqan2, 100);
```